### ISSUE TITLE
A test failure with a moose submodule update

### ISSUE THREAD

**Issue by YaqiWang** [idaholab/virtual_test_bed#215]
The failure can be seen here: https://civet.inl.gov/job/1435246/. The test is using PJFNKMO, without using `constant_matrices = true`. I suspect the recent change I made in MOOSE (https://github.com/idaholab/moose/pull/23873) is causing this. The tolerance of that test is not very tight and the CSV diff is not high. If we confirm indeed it is that change causing the failure. I think we can safely regold the test.

**Comment by GiudGiud** [comment]
Would recomputing the matrices too often have this effect? The matrices should have been constant every time?

anyhow I think looking at after / before the PR is enough to diagnose here, and regold is indeed the way to go

**Comment by YaqiWang** [comment]
Evaluating matrices on every linear iteration is not good for sure in my opinion. I just verified that undoing the change, I can get
```
Users/wangy2/projects/gitlib/virtual_test_bed/htgr/pbmr400/steady.neutronics_only ........... RUNNING [82.73s]
```
I think we should just regold the file. I will tight the tolerance and do two runs again to make sure they will eventually converge to the same.

**Comment by GiudGiud** [comment]
sounds good.
Careful with the runtime if you tighten the tolerance, that s usually why I limited it

**Comment by YaqiWang** [comment]
I will not use the tight tolerance in the PR but will just to verify. I will also report the cpu time of the two runs. The one with fix should run much faster.

**Comment by GiudGiud** [comment]
Closing as addressed by #217
